### DISCUSSION TITLE
Implement Long Press Functionality to Edit Document Information

### DISCUSSION THREAD

**Discussion by fire1ce** [LeoKlaus/Paperparrot#151] [Feature Requests]
I would like to suggest an enhancement to the PAPERPARROT application, specifically targeting the user interaction with documents. Currently, there is an option to toggle the "view on tap" feature, which opens a preview window of the selected document using iOS’s native preview functionality.

While this feature is immensely helpful, I believe the addition of a long press functionality to edit document information directly from the document list view would significantly improve user experience and efficiency. This would provide a quick and intuitive way for users to update document details without navigating away from the main document list.

Suggested Implementation:
A long press on any document in the list view could trigger an edit mode, allowing users to update document information such as title, tags, and other metadata.
Visual feedback and a concise menu could be provided to ensure users are aware of the available actions and how to execute them.
This enhancement would streamline the document management process within PAPERPARROT, providing a more seamless and user-friendly experience.

**Comment by LeoKlaus** [comment]
Paperparrot already has a context menu for small preview items (like those in the main documents view and search results). You can open it by long pressing (or right clicking on Mac) on the preview.

The context menu does allow for some quick changes like adding a tag or setting a correspondent. More complex interactions like changing the title can only be done within the actual editing view, which can be opened by tapping "Edit Document" in the context menu.
![Simulator Screenshot - iPhone 15 Pro - 2023-11-01 at 11 06 46](https://github.com/LeoKlaus/Paperparrot/assets/66492595/d69b7740-c1e2-4a42-82ba-5664672803e5)

**Reply by fire1ce** [reply]
I was referring to the iOS app.

**Reply by LeoKlaus** [reply]
Me too, this screenshot is from the iOS app.
Try long pressing on one of your documents in the overview. If you have the list view active, make sure to long press on either the title or preview image, the blank space doesn't register touches.

**Reply by fire1ce** [reply]
Oh, you are right. Was pressing on the "empty" white space. 
Closing this discussion too. 
Thanks again, keep the good work!

**Reply by LeoKlaus** [reply]
It's unfortunate but that's how Swiftui works. I haven't found a way yet to make the empty area respond to touches as well. 
Thank you for the reply!
